### DISCUSSION TITLE
Shadow softness blur

### DISCUSSION THREAD

**Discussion by Lemovision** [google/model-viewer#2907] [General]
Hi is it possible to improve the shadow blurring method?
It seems the shadowmap you use has good resolution because it looks good with zero shadow softness
However when using shadow softness it becomes very "blocky" instead of actually soft

![shadow](https://user-images.githubusercontent.com/61175245/138357124-2423ace3-e321-448e-8893-a5109abc2264.gif)

I admit I made it darker than it should for demo here, and in most cases it's not as visible
But I can still see the blockyness with light grey shadow, and especially for animation as the blockyness flickers
I suppose you're using a very fast box blur for performance here
Still is it possible to have some gaussian blur or at least a tiny bit smoother blur?
Or too much impact on performance?

**Comment by elalish** [comment]
Yes, indeed our shadows are very fast and very simple. In fact what happens as you increase softness is we simply decrease the shadow resolution (making it even faster), hence the blockiness. In general we push for performance, since we keep MV operational across the broadest range of devices (even low-end mobile). We chose this because as you said, it's mostly hard to notice. Still, phones are getting faster all the time and at some point we should probably switch to a higher quality method. Of course keep in mind that our shadow currently  doesn't even have anything to do with our lighting  😂

**Reply by Lemovision** [reply]
Alright that's fair enough, thanks for the info

Or maybe there could be a way to adjust the shadow resolution dynamically depending on the device performance?
It seems to me you're already doing something like that with the render scale
No idea how that works, but maybe you could adjust shadow resolution after render scale is 1 and if frame rate is still good?
